java/src/main/java/com/powsybl/dataframe/network/adders/ThreeWindingsTransformerSeries.java

```diff
@@ -0,0 +1,140 @@
+    private final StringSeries voltageLevels3;
+    private final StringSeries connectableBuses3;
+    private final StringSeries buses3;
+    private final IntSeries nodes3;
+
+    ThreeWindingsTransformerSeries(UpdatingDataframe dataframe) {
+        super(dataframe);
+        this.ratedU0 = dataframe.getDoubles("rated_u0");
+        this.r1 = dataframe.getDoubles("r1");
+        this.x1 = dataframe.getDoubles("x1");
+        this.g1 = dataframe.getDoubles("g1");
+        this.b1 = dataframe.getDoubles("b1");
+        this.ratedU1 = dataframe.getDoubles("rated_u1");
+        this.ratedS1 = dataframe.getDoubles("rated_s1");
+        this.r2 = dataframe.getDoubles("r2");
+        this.x2 = dataframe.getDoubles("x2");
+        this.g2 = dataframe.getDoubles("g2");
+        this.b2 = dataframe.getDoubles("b2");
+        this.ratedU2 = dataframe.getDoubles("rated_u2");
+        this.ratedS2 = dataframe.getDoubles("rated_s2");
+        this.r3 = dataframe.getDoubles("r3");
+        this.x3 = dataframe.getDoubles("x3");
+        this.g3 = dataframe.getDoubles("g3");
+        this.b3 = dataframe.getDoubles("b3");
+        this.ratedU3 = dataframe.getDoubles("rated_u3");
+        this.ratedS3 = dataframe.getDoubles("rated_s3");
+        this.voltageLevels1 = dataframe.getStrings("voltage_level1_id");
+        this.connectableBuses1 = dataframe.getStrings("connectable_bus1_id");
+        this.buses1 = dataframe.getStrings("bus1_id");
+        this.nodes1 = dataframe.getInts("node1");
+        this.voltageLevels2 = dataframe.getStrings("voltage_level2_id");
+        this.connectableBuses2 = dataframe.getStrings("connectable_bus2_id");
+        this.buses2 = dataframe.getStrings("bus2_id");
+        this.nodes2 = dataframe.getInts("node2");
+        this.voltageLevels3 = dataframe.getStrings("voltage_level3_id");
+        this.connectableBuses3 = dataframe.getStrings("connectable_bus3_id");
+        this.buses3 = dataframe.getStrings("bus3_id");
+        this.nodes3 = dataframe.getInts("node3");
+    }
+
+    ThreeWindingsTransformerAdder create(Network network, int row) {
+        String id = ids.get(row);
+        VoltageLevel vl1 = getVoltageLevelOrThrow(network, voltageLevels1.get(row));
+        VoltageLevel vl2 = getVoltageLevelOrThrow(network, voltageLevels2.get(row));
+        VoltageLevel vl3 = getVoltageLevelOrThrow(network, voltageLevels3.get(row));
+
+        Substation s1 = vl1.getSubstation().orElseThrow(() -> new PowsyblException(COULD_NOT_CREATE_TRANSFORMER + id + NO_SUBSTATION));
+        Substation s2 = vl2.getSubstation().orElseThrow(() -> new PowsyblException(COULD_NOT_CREATE_TRANSFORMER + id + NO_SUBSTATION));
+        Substation s3 = vl3.getSubstation().orElseThrow(() -> new PowsyblException(COULD_NOT_CREATE_TRANSFORMER + id + NO_SUBSTATION));
+        if (!(s1 == s2 && s1 == s3)) {
+            throw new PowsyblException(COULD_NOT_CREATE_TRANSFORMER + id + ": both voltage ids must be on the same substation");
```

Comment:
I think "all voltage ids" is better than "both voltage ids" in the error since there are more than two here